src/NuGet.Clients/VisualStudio/Extensibility/IVsPackageProjectMetadata.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Runtime.InteropServices;
+
+namespace NuGet.VisualStudio
+{
+    /// <summary>
+    /// contain information about project where we're executing nuget actions
+    /// </summary>
+    [ComImport]
+    [Guid("7ce0caae-1b99-4cd1-80bf-348972975bcf")]
+    public interface IVsPackageProjectMetadata
+    {
+        /// <summary>
+        /// guid for batch events of the project
+        /// </summary>
+        string Guid { get; }
```

Comment:
> and besides you have to provide a guid while declaring Com class

Users don't use guid declared on this class, they reference the assembly and it is done automatically for them. The guid tracks the interface across different assemblies so that it can match them up.

This property on the other hand returns the a guid that is randomly generated for the nuget project each time the solution is opened, there's no value for the user.

This could be named BatchId and given a unique guid that corresponds to that single operation. I'm just not sure how Guid is supposed to be used in this implementation, and the name isn't helpful.